tests/ExamplesTest.php

```diff
@@ -0,0 +1,198 @@
+
+update command succeeded
+reply: { "n" : 2, %s }
+
+find command started
+command: { "find" : "coll", "filter" : {  }, "batchSize" : 2, "$db" : "test", "lsid" : { %s }%S }
+
+find command succeeded
+reply: { "cursor" : { "firstBatch" : [ { "_id" : { "$oid" : "%s" }, "x" : 1 }, { "_id" : { "$oid" : "%s" }, "x" : 2, "y" : 1 } ], "id" : %d, "ns" : "test.coll" }, %s }
+
+{ "_id" : { "$oid" : "%s" }, "x" : 1 }
+{ "_id" : { "$oid" : "%s" }, "x" : 2, "y" : 1 }
+getMore command started
+command: { "getMore" : %d, "collection" : "coll", "batchSize" : 2, "$db" : "test", "lsid" : { %s }%S }
+
+getMore command succeeded
+reply: { "cursor" : { "nextBatch" : [ { "_id" : { "$oid" : "%s" }, "x" : 3, "y" : 1 } ], "id" : 0, "ns" : "test.coll" }, %s }
+
+{ "_id" : { "$oid" : "%s" }, "x" : 3, "y" : 1 }
+OUTPUT;
+
+        yield 'command_logger' => [
+            'file' => __DIR__ . '/../examples/command_logger.php',
+            'expectedOutput' => $expectedOutput,
+        ];
+
+        $expectedOutput = <<<'OUTPUT'
+object(MongoDB\Examples\PersistableEntry)#%d (%d) {
+  ["id":"MongoDB\Examples\PersistableEntry":private]=>
+  object(MongoDB\BSON\ObjectId)#%d (%d) {
+    ["oid"]=>
+    string(24) "%s"
+  }
+  ["name"]=>
+  string(7) "alcaeus"
+  ["emails"]=>
+  array(2) {
+    [0]=>
+    object(MongoDB\Examples\PersistableEmail)#%d (%d) {
+      ["type"]=>
+      string(4) "work"
+      ["address"]=>
+      string(19) "alcaeus@example.com"
+    }
+    [1]=>
+    object(MongoDB\Examples\PersistableEmail)#%d (%d) {
+      ["type"]=>
+      string(7) "private"
+      ["address"]=>
+      string(18) "secret@example.com"
+    }
+  }
+}
+OUTPUT;
+
+        yield 'persistable' => [
+            'file' => __DIR__ . '/../examples/persistable.php',
+            'expectedOutput' => $expectedOutput,
+        ];
+
+        $expectedOutput = <<<'OUTPUT'
+object(MongoDB\Examples\TypeMapEntry)#%d (%d) {
+  ["id":"MongoDB\Examples\TypeMapEntry":private]=>
+  object(MongoDB\BSON\ObjectId)#%d (%d) {
+    ["oid"]=>
+    string(24) "%s"
+  }
+  ["name":"MongoDB\Examples\TypeMapEntry":private]=>
+  string(7) "alcaeus"
+  ["emails":"MongoDB\Examples\TypeMapEntry":private]=>
+  array(2) {
+    [0]=>
+    object(MongoDB\Examples\TypeMapEmail)#%d (%d) {
+      ["type":"MongoDB\Examples\TypeMapEmail":private]=>
+      string(4) "work"
+      ["address":"MongoDB\Examples\TypeMapEmail":private]=>
+      string(19) "alcaeus@example.com"
+    }
+    [1]=>
+    object(MongoDB\Examples\TypeMapEmail)#%d (%d) {
+      ["type":"MongoDB\Examples\TypeMapEmail":private]=>
+      string(7) "private"
+      ["address":"MongoDB\Examples\TypeMapEmail":private]=>
+      string(18) "secret@example.com"
+    }
+  }
+}
+OUTPUT;
+
+        yield 'typemap' => [
+            'file' => __DIR__ . '/../examples/typemap.php',
+            'expectedOutput' => $expectedOutput,
+        ];
+    }
+
+    public function testChangeStream(): void
```

Comment:
This test (and `testWithTransaction`) below was extracted as it needs to check whether change streams are supported. I initially wanted to provide a closure in the data provider, but that causes errors due to closures not being serialisable (which is required to run tests in isolation). Instead of passing a string callable I opted for extracting the tests from the data provider and calling the original test method manually instead.